content/en/os/1.14.x/login/regaining-access/index.markdown

```diff
@@ -83,4 +83,145 @@ The Bottlerocket node should be accessible again.
 
 ## Steps to Regain Access on ECS
 
-Bottlerocket supports `ecs exec` starting in v1.14.0, so a similar solution is possible.
+You can regain access to a Bottlerocket node using ECS Exec starting in v1.14.0.
+
+This solution requires the use of [ECS Exec](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html) and [associated IAM permissions](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html#ecs-exec-enabling-and-using).
+The instructions for proper IAM permissions to enable ECS Exec can be found in the AWS blog post [Using Amazon ECS Exec to access your containers on AWS Fargate and Amazon EC2](https://aws.amazon.com/blogs/containers/new-using-amazon-ecs-exec-access-your-containers-fargate-ec2/).
+
+1. First, you will need to setup a few environment variables. `CONTAINER_NAME` and `SERVICE_NAME` are not specifically relevant, but they need to be consistent across multiple files and commands.
+
+```shell
+export CONTAINER_NAME="regain-access"
+export CLUSTER=<your cluster name>
+export SERVICE_NAME=regain-access
+export TASK_ROLE_ARN=<the ARN for your IAM task role>
+```
+
+2. Now you will create a task definition that will mount the Bottlerocket `apiclient` and the related socket path as well as giving it the proper SELinux labels.
+Piping `cat` to a file to generate the task definition[^1] JSON will allow for interpolating the variables into the final file.
+
+```shell
+cat << EOF > task-def.json 
+{
+    "family": "regain-access",
+    "containerDefinitions": [
+        {
+            "name": "$CONTAINER_NAME",
+            "image": "fedora",
```

Comment:
Up to you, but should you use something smaller, like `alpine`. Or `al2023`. It doesn't really matter, since `apiclient` is statically-linked so it doesn't require anything from the container image.

Reply:
This is just parity with the related instructions in k8s. There is also a footnote

> This task definition uses `fedora` as an image, but almost any base image with a shell will also work.